src/System.Management.Automation/utils/tracing/SysLogProvider.cs

```diff
@@ -0,0 +1,543 @@
+        /// <param name="args">An array of zero or more payload objects</param>
+        private static void GetEventMessage(StringBuilder sb, PSEventId eventId, params object[] args )
+        {
+            int parameterCount;
+            string resourceName = EventResource.GetMessage((int) eventId, out parameterCount);
+
+            if (resourceName == null)
+            {
+                // If an event id was specified that is not found in the event resource lookup table,
+                // use a placeholder message that includes the event id.
+                resourceName = EventResource.GetMissingEventMessage(out parameterCount);
+                Diagnostics.Assert(false, sb.ToString());
+                args = new object[] {eventId};
+            }
+
+            string resourceValue = GetResourceString(resourceName);
+            if (parameterCount > 0)
+            {
+                sb.AppendFormat(resourceValue, args);
+            }
+            else
+            {
+                sb.Append(resourceValue);
+            }
+        }
+
+        #region logging
+
+        // maps a LogLevel to an associated SysLogPriority
+        static NativeMethods.SysLogPriority[] _levels =
+        {
+            NativeMethods.SysLogPriority.Info,
+            NativeMethods.SysLogPriority.Critical,
+            NativeMethods.SysLogPriority.Error,
+            NativeMethods.SysLogPriority.Warning,
+            NativeMethods.SysLogPriority.Info,
+            NativeMethods.SysLogPriority.Info
+        };
+
+        /// <summary>
+        /// Logs a activity transfer
+        /// </summary>
+        /// <param name="parentActivityId">The parent activity id.</param>
+        public void LogTransfer(Guid parentActivityId)
+        {
+            // NOTE: always log
+            int threadId = Thread.CurrentThread.ManagedThreadId;
+            string message = string.Format(CultureInfo.InvariantCulture,
+                                           "({0:X}:{1:X}:{2:X}) [Transfer]:{3} {4}",
+                                           _commitId, threadId, PSChannel.Operational,
+                                           parentActivityId.ToString("B"),
+                                           Activity.ToString("B"));
+
+            NativeMethods.SysLog(NativeMethods.SysLogPriority.Info, message);
+        }
+
+        /// <summary>
+        /// Logs the activity identifier for the current thread.
+        /// </summary>
+        /// <param name="activity">The Guid activity identifier</param>
+        public void SetActivity(Guid activity)
+        {
+            int threadId = Thread.CurrentThread.ManagedThreadId;
+            Activity = activity;
+
+            // NOTE: always log
+            string message = string.Format(CultureInfo.InvariantCulture,
+                                           "({0:X}:{1:X}:{2:X}) [Activity] {3}",
+                                           _commitId, threadId, PSChannel.Operational, activity.ToString("B"));
```

Comment:
We use Activity.ToString("B") many times - I think it makes sense to keep it as a string.

Reply:
I prefer not to change this; it will require another thread local storage value as well as another accessor (see what Activity.get needs to do).  Usage is limited to setting and transferring activities.

Reply:
Closed.